mono/metadata/appdomain.c

```diff
@@ -1405,26 +1406,23 @@ mono_domain_fire_assembly_load (MonoAssembly *assembly, gpointer user_data)
 		g_assert (assembly_load_field);
 	}
 
-	mono_field_get_value ((MonoObject*) domain->domain, assembly_load_field, &load_value);
-	if (load_value == NULL) {
-		/* No events waiting to be triggered */
-		goto leave;
-	}
+	mono_field_get_value_handle (appdomain, assembly_load_field, &load_value);
```

Comment:
So there's a problem here: System.AppDomain.AssemblyLoad is an event delegate.  So loading from it gives back a managed pointer.  Which is completely obscured by the signature of `mono_field_get_value_handle` which just has a `gpointer *` out arg.

You should really use `MONO_HANDLE_NEW_GET_FIELD (appdomain, MonoObject, assembly_load_field)` here.

Reply:
Yeah I saw a void* and was wondering.

Reply:
We only check  the nullness. MONO_HANDLE_NEW_GET_FIELD does actually look more efficient maybe. What type to use for `load_value`?